spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/integration/inbound/PubSubMessageSource.java

```diff
@@ -91,14 +77,11 @@ protected Object doReceive() {
 		List<ConvertedAcknowledgeablePubsubMessage> messages
 				= this.pubSubSubscriberOperations.pullAndConvert(this.subscriptionName, 1, true, this.payloadType);
 
-		if (messages.size() < 1) {
+		if (messages.isEmpty()) {
 			return null;
 		}
 
-		if (messages.size() > 1) {
-			LOGGER.warn("One message expected; " + messages.size() + " received. Returning the first one.");
-		}
-
+		Assert.isTrue(messages.size() == 1, "Only one Pub/Sub message expected.");
```

Comment:
`isTrue()` doesn't fit our requirements. It throws an `IllegalArgumentException`, which doesn't reflect the situation. That's why I recommended to use a `state()` here.